pkg/kn/commands/service/create.go

```diff
@@ -260,3 +267,32 @@ func constructService(cmd *cobra.Command, editFlags ConfigurationEditFlags, name
 	}
 	return &service, nil
 }
+
+// constructServiceFromFile creates struct from provided file
+func constructServiceFromFile(cmd *cobra.Command, editFlags ConfigurationEditFlags, name, namespace string) (*servingv1.Service, error) {
+	var service servingv1.Service
+	file, err := os.Open(editFlags.File)
+	if err != nil {
+		return nil, err
+	}
+	decoder := yaml.NewYAMLOrJSONDecoder(file, 512)
+
+	err = decoder.Decode(&service)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I can comment these lines and pass all tests. Add a test for bad input file.

Reply:
Done in https://github.com/knative/client/pull/913/commits/9a3c1b92bd85e16386ee9613296481877f774cb6